### PR TITLE
Update filter when programmatically changing quick input value

### LINKED ISSUE #137279: QuickPick: filter not reset on value set programmatically
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 1.62.2
- OS Version: Darwin x64 20.6.0

Steps to Reproduce:

1. Setup a QuickPick as follows
```typescript
const qp = window.createQuickPick();
qp.items = [
	{ label: "This is first" },
	{ label: "This is second" },
];
const clearIcon = { iconPath: new ThemeIcon('trash') };
qp.buttons = [clearIcon];
const disposables = [
	qp.onDidTriggerButton((button) => {
		if (button === clearIcon) {
			qp.value = "";
		}
	}),
	qp
];
qp.onDidHide(() => {
	for (const d of disposables) {
		d.dispose();
	}
});
qp.show();
```

2. Run the code above
3. When the quickpick is shown, type `first`
4. The menu should only show one item because of the filter word `first`
5. Click the trash button to clear the value programmatically
6.

ACTUAL: The filter doesn't update, and only show one menu items
EXPECTED: The filter updates, and show two menu items

https://user-images.githubusercontent.com/3455662/141921763-0521887c-ef66-4280-8db4-c1b5cd7b2e17.mov

### PR DESCRIPTION
**Pull request by bpasero**
Continuation from https://github.com/microsoft/vscode/pull/137666

This PR fixes #137279

### CODE REVIEW

**Comment by bpasero** [comment]
@TylerLeonhardt leaving this to you for now. I suggest you also look into restoring the tests you had in your previous changes to see if they pass with this change.

**Review by chrmarti — APPROVED** [review]

File: src/vs/base/parts/quickinput/browser/quickInput.ts [hunk 1]
```diff
@@ -487,9 +487,19 @@ class QuickPick<T extends IQuickPickItem> extends QuickInput implements IQuickPi
...
+
+	private doSetValue(value: string, skipUpdate?: boolean): void {
 		if (this._value !== value) {
-			this._value = value || '';
-			this.update();
+			this._value = value;
+			if (!skipUpdate) {
+				this.update();
+			}
+			const didFilter = this.ui.list.filter(this.filterValue(this._value));
```
Comment (chrmarti) [inline src/vs/base/parts/quickinput/browser/quickInput.ts:499]: The UI should only be updated if this QuickPick is visible (another QuickPick could be visible).

Comment (bpasero) [reply src/vs/base/parts/quickinput/browser/quickInput.ts]: @chrmarti 👍 , see https://github.com/microsoft/vscode/pull/137857/commits/f00d30357fb835e272e6e244b836a9e235482486

**Review by raghavthind2005 — APPROVED** [review]

**Review by TylerLeonhardt — APPROVED** [review]

**Comment by TylerLeonhardt** [comment]
Thanks @bpasero!

**Pull request merged**